doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,534 @@
+    commandlines
+13. A user wants to open the Windows Terminal with multiple tabs and panes open
+    simultaneously, each with different profiles, starting directories, even
+    commandlines, and specific split sizes
+14. A user wants to use a file to provide a reusable startup configuration with
+    many steps, to avoid needing to type the commandline each time.
+
+## Solution Design
+
+### Proposal 1 - Parameters
+
+Initially, I had considered arguments in the following style:
+
+* `--help`: Display the help message
+* `--version`: Display version info for the Windows Terminal
+* `--list-profiles`: Display a list of the available profiles
+  - `--all` to also show "hidden" profiles
+  - `--verbose`? To also display GUIDs?
+* `--open-settings`: Open the settings file
+* `--profile <profile name>`: Start with the given profile, by name
+* `--guid <profile guid>`: Start with the given profile, by GUID
+* `--startingDirectory <path>`: Start in the given directory
+* `--initialRows <rows>`, `--initialCols <rows>`: Start with a specific size
+* `--initialPosition <x,y>`: Start at an initial location on the screen
+* `-- <commandline>`: Start with this commandline instead
+
+However, this style of arguments makes it very challenging to start multiple
+tabs or panes simultaneously. How would a user start multiple panes, each with a
+different commandline? As configurations become more complex, these commandlines
+would quickly become hard to parse and understand for the user.
+
+### Proposal 2 - Commands and Parameters
+
+Instead, we'll try to seperate these arguments by their responsibilities. Some
+of these arguments cause something to happen, like `help`, `version`, or
+`open-settings`. Other arguments act more like modifiers, like for example
+`--profile` or `--startingDirectory`, which provide additional information to
+the action of _opening a new tab_. Lets try and define these concepts more
+clearly.
+
+**Commands** are arguments that cause something to happen. They're provided in
+`kebab-case`, and can have some number of optional or required "parameters".
+
+**Parameters** are arguments that provide additional information to "commands".
+They can be provided in either a long form or a short form. In the long form,
+they're provided in `--camelCase`, with two hyphens preceeding the argument
+name. In short form, they're provided as just a single character preceeded by a
+hyphen, like so: `-c`.
+
+Let's enumerate some possible example commandlines, with explanations, to
+demonstrate:
+
+### Sample Commandlines
+
+```sh
+
+# Runs the user's "Windows Powershell" profile in a new tab (user story 2)
+wt new-tab --profile "Windows Powershell"
+wt --profile "Windows Powershell"
+wt -p "Windows Powershell"
+
+# Runs the user's default profile in a new tab, running cmd.exe (user story 3)
+wt cmd.exe
+
+# display the help text (user story 4)
+wt help
+wt --help
+wt -h
+wt -?

```

Comment:
Windows users are familiar with the `/` for parameters. Even if none of the others take `/` because we're standardizing on the unix-y way of using `-`, I think we should min-bar still accept `/?` to open the help text.

Reply:
that's a fair point. I can add that.

Reply:
lol apparently I just added it in the impl, but not the spec 😝